### DISCUSSION TITLE
SE3 enviroment not working

### DISCUSSION THREAD

**Discussion by mateoschaerer** [RosettaCommons/RFdiffusion#154] [General]
Hello recently I used the code outlined in the README to use the RFdiffusion model locally (and had no issues)  but if I run the code now  there seems to an issue with the SE3 enviroment not being built correctly

**Comment by chasekosborne** [comment]
This is a pretty common problem, this may depend on the system you have downloaded and what Cudatoolkit version you are running on your system. If this is the case, the solution for (somewhat) newer versions is listed [here](https://github.com/RosettaCommons/RFdiffusion/issues/14#issuecomment-1505150974). You can check what cudatoolkit version you have using the command:
`nvcc --version`

If this solution doesn't work for you just let me know, there are workarounds for this problem past this quick fix so long as you have sudo rights on your system.

**Comment by roccomoretti** [comment]
If you're dealing with CUDA versioning, I recommend reading [this post](https://stackoverflow.com/a/53504578) to help keep a few things straight.

Note that `nvcc --version` will tell you info about the CUDA runtime that's installed when you installed the nvcc compiler. If you install a CUDA runtime with Conda (e.g. you're installing the cudatoolkit package), you probably won't get the nvcc compiler installed with it, so the CUDA runtime version you're using inside the Conda environment won't necessarily correspond to the CUDA runtime version reported by nvcc. 

And if you are installing the CUDA runtime with Conda, it doesn't really matter which CUDA runtime version is installed outside of Conda. Instead what matters is what CUDA versions the driver can support. The `nvidia-smi` command should tell you the maximum version of CUDA which the GPU driver can support, and you can (usually) use Conda to install a version of the CUDA runtime/cudatoolkit that fits within that maximum. (Even if it's more recent than the version reported by nvcc outside of Conda.)  A runtime which has a lower version than the driver (usually) works, but a runtime newer than the driver won't.
